CenterForOpenScience/ember-osf-web
# [No ticket] Overview subject search

-   Ticket: [No ticket]
-   Feature flag: n/a

## Purpose
- Fix the registries overview's subject search widget

## Summary of Changes
- Load the registration provider before querying its subjects
- Update types accordingly

## Screenshot(s)
![image](https://user-images.githubusercontent.com/51409893/139326358-93fba83c-dd9c-4441-acd9-477df4bf665b.png)

## Side Effects
- None

## QA Notes
- The subject search widget on the registration overview page should work as-expected now
  - Registration admins editing the Registration's Metadata using the Metadata panel on the right should be able to search for subjects and apply those subjects onto the registration
